File: ActivityRouterProject/app/src/main/java/android/stack/activityrouterproject/MainActivity.java
package android.stack.activityrouterproject;

import android.os.Bundle;
import android.stack.router_annotation.Route;
import android.stack.router_api.Router;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

/**
 * @author zhangchaozhou
 */
@Route("/MainActivity")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    public void listen(View view) {
        Router.getInstance(this).startActivity("/SpeakActivity");
    }

    public void speak(View view) {
        Router.getInstance(this).startActivity("/ListenActivity");
    }
}

File: ViewBinding/app/src/main/java/androidstack/viewbinding/MainActivity.kt
package androidstack.viewbinding

import android.os.Bundle
import androidstack.viewbinding.databinding.ActivityMainBinding
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //获取绑定类实例
        binding = ActivityMainBinding.inflate(layoutInflater)
        //通过绑定类获取布局后，传给setContentView设置内容视图
        setContentView(binding.root)
        //通过绑定类获取对应的ID视图，进行操作
        binding.tvHelloWorld.text = "视图绑定"
    }
}